pkg/virt-handler/vm_test.go

```diff
@@ -377,6 +377,9 @@ var _ = Describe("VirtualMachineInstance", func() {
 		It("should update from Scheduled to Running, if it sees a running Domain", func() {
 			vmi := v1.NewMinimalVMI("testvmi")
 			vmi.UID = testUUID
+			vmi.Spec.Domain.CPU = &v1.CPU{
```

Comment:
I have seen that new integration tests have been added, which is good. I'd also add new unit tests to make the logic verifiable more easily, so as to i.e. support refactorings without having to run the integration tests to verify.